services/iaasalpha/wait/wait.go

```diff
@@ -165,3 +176,59 @@ func DeleteServerWaitHandler(ctx context.Context, a APIClientInterface, projectI
 	handler.SetTimeout(20 * time.Minute)
 	return handler
 }
+
+// ProjectRequestWaitHandler will wait for a request to succeed
+// It receives a request id that can be obtained from the x-request-id header in the http response of any operation in the IaaS API.
+// To get the raw http response of an operation using the SDK, use the runtime.WithCaptureHTTPResponse method from the core pkg.
+// Then the value of the request id can be obtained by accessing the header key which is defined in the constant XRequestIDHeader in this package.
+// Example usage:
```

Comment:
Not sure if this is too much in a method description. We will also have examples but I thought about making it clear here how to use everything together

Reply:
I agree it makes sense to have this detailed description :+1: